crates/phactory/src/system/mod.rs

```diff
@@ -860,13 +860,19 @@ impl<Platform: pal::Platform> System<Platform> {
                 .expect("empty master key history")
                 .secret,
         );
-        let gatekeeper = gk::Gatekeeper::new(
+        let mut gatekeeper = gk::Gatekeeper::new(
             master_key_history,
             block.recv_mq,
             block
                 .send_mq
                 .channel(MessageOrigin::Gatekeeper, master_key.into()),
         );
+        if let Some(params) = block.storage.tokenomic_parameters() {
```

Comment:
Does it update the tokenomic parameter at every block?

Reply:
No, it's a one-shot operation on the GK launch. Later updates will keep using mq.